server/src/test/java/org/elasticsearch/index/mapper/KeywordFieldMapperTests.java

```diff
@@ -381,6 +391,54 @@ public void testNormalizer() throws IOException {
         assertThat(fieldType.indexOptions(), equalTo(IndexOptions.NONE));
         assertEquals(DocValuesType.SORTED_SET, fieldType.docValuesType());
     }
+    
+    // Test that a name clash with a custom normalizer will favour the index's normalizer rather than the out-of-the-box 
+    // one of the same name. (However this "feature" will be removed with https://github.com/elastic/elasticsearch/issues/22263 ) 
+    public void testNameClashNormalizer() throws IOException {
```

Comment:
I think it'd make sense for this check to live in `AnalysisRegistryTests`, because it is really testing our analyzer registration behavior instead of anything specific to the keyword field mapper. I took a look and it seems straightforward to add a test there.

Reply:
Isn't this more related to the interaction of AnalysisRegistry and an instance of an index though? That's where the ambiguity comes in - will the field mapping pick the "local" normalizer defined in the index settings or the pre-configured one we now ship with? 
That's why I thought it made sense to test it in the context of something like KeywordFieldMapper - we want to create an index with a field to see if it makes the correct choices.

Happy to put something in AnalysisRegistry if possible but I thought adding  the index creation stuff might be hard.